### PR TITLE
Fix #7307: Add WaterClass to MP_TREES

### LINKED ISSUE #7307: Inconsistent detection of coast tiles
## Version of OpenTTD

c700b0cbb9de1b7a1277931eb1dfed55c6bd564d

## Expected result

Consistent detection of coast tiles when placing newgrf industry.

## Actual result

Frequently, but inconsistently, coast tiles aren't detected correctly.

Bulldozing the tiles resolves the issue.

## Steps to reproduce

Try building a Bulk Terminal in FIRS 3, along a straight run of SE coast (easiest placement).

Occasionally this won't be permitted.  Then bulldoze the 4 coast tiles for the location.  Then building will be permitted.  

This might be due to some other tile var, not coast detection.  Or it could easily be a bug in FIRS or nmlc.  But it looks to me like some tile prop isn't set correctly. 

Been present for a long time, causes issues placing these and similar industries at map-gen.

coast_bug_1
coast_bug_2
coast_bug_3

### PR DESCRIPTION
**Pull request by Eddi-z**
Regression seems to fail on some money changes (maybe unwanted side effects?)

### CODE REVIEW

File: src/tree_map.h [hunk 1]
```diff
@@ -133,6 +134,7 @@ static inline void SetTreeGroundDensity(TileIndex t, TreeGround g, uint d)
 	assert(IsTileType(t, MP_TREES)); // XXX incomplete
 	SB(_m[t].m2, 4, 2, d);
 	SB(_m[t].m2, 6, 3, g);
+	SetWaterClass(t, g==TREE_GROUND_SHORE?WATER_CLASS_SEA:WATER_CLASS_INVALID);
```
Comment (PeterN) [inline src/tree_map.h]: Spaces please!

**Comment by PeterN** [comment]
Why do trees need a water class?

**Comment by Eddi-z** [comment]
Because trees can be on the shore, and that switches the tile type to MP_TREES. Those "trees on shore" should be detected as sea by anything that cares.

**Comment by PeterN** [comment]
Can you please update landscape.html and landscape_grid.html?

**Review by PeterN — APPROVED** [review]

**Pull request merged**